.github/workflows/go-test.yaml

```diff
@@ -19,15 +19,15 @@ jobs:
           path: src/github.com/nats-io/nats-server
 
       - name: Setup Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v4
         with:
           go-version: ${{matrix.go}}
 
       - name: Install deps
         shell: bash --noprofile --norc -x -eo pipefail {0}
         run: |
-          go get -u honnef.co/go/tools/cmd/staticcheck
-          go get -u github.com/client9/misspell/cmd/misspell
+          go install honnef.co/go/tools/cmd/staticcheck@latest
+          go install github.com/client9/misspell/cmd/misspell@latest
```

Comment:
I think we should remove these and use `golangci-lint`, as we already do in `runTestsOnTravis.sh`. That way the single source of linting truth is `.golangci.yml` for all.

Reply:
golangci includes the misspell too?

Reply:
It does

Reply:
sounds good to me, ok changing to this:

```sh
      - name: Install deps
        shell: bash --noprofile --norc -x -eo pipefail {0}
        run: |
          go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.53.3

      - name: Lint
        shell: bash --noprofile --norc -x -eo pipefail {0}
        run: |
          golangci-lint run
```